src/apps/tools.clj

```diff
@@ -96,19 +108,24 @@
   [{:keys [tools]}]
   (transaction
     (let [tool-ids (doall (map add-new-tool tools))]
+      (dorun (map permissions/register-public-tool tool-ids))
       {:tool_ids tool-ids})))
 
 (defn update-tool
-  [overwrite-public {:keys [id container] :as tool}]
+  [user overwrite-public {:keys [id container] :as tool}]
   (persistence/update-tool tool)
   (when container
     (set-tool-container id overwrite-public container))
-  (get-tool id))
+  (get-tool user id))
 
 (defn delete-tool
   [user tool-id]
-  (let [{:keys [name location version]} (get-tool tool-id)]
+  (let [{:keys [name location version]} (get-tool user tool-id)]
     (validate-tool-not-used tool-id)
     (log/warn user "deleting tool" tool-id name version "@" location))
   (delete tools (where {:id tool-id}))
+  (try+
+    (permissions/delete-tool-resource tool-id)
+    (catch [:status 404] _
+      (log/warn "tool resource" tool-id "not found by permissions service")))
   nil)
```

Comment:
Ah, no problem. It's not really obvious from inside this namespace 😁 

This also applies for the update endpoint. Once the features are added for private tool support, there will be endpoints specific to private tools that will check permissions.

I think for now I'm leaning towards not allowing admins to update or delete private tools. It shouldn't be too much work to add that support later if we decide we need it.